client/components/application-summary.js

```diff
@@ -205,6 +245,117 @@ const ApplicationSummary = () => {
     return <HoldingPage />;
   }
 
+
+/**
+ * Recursively find all fields within a section and compare their values.
+ * @param {object} props - Contains all saved and current data (savedValues, currentValues, initialValues).
+ * @param {object} sectionData - The section object containing fields and nested subsections.
+ * @returns {boolean} - True if any field in the section differs, false otherwise.
+ */
+const hasSectionChangedDeep = (props, sectionData) => {
+  const { savedValues, currentValues, initialValues } = props;
+
+  if (!savedValues || !currentValues || !initialValues) {
+    console.error("Missing savedValues, currentValues, or initialValues in props.");
+    return false;
+  }
+
+  /**
+   * Recursive helper function to compare values.
+   * @param {object} data - The section or field data to process.
+   * @returns {boolean} - True if any field differs, false otherwise.
+   */
+  const compareFields = (data) => {
+    let changed = false;
+
+    Object.entries(data).forEach(([key, value]) => {
+      // Handle nested objects (e.g., subsections)
+      if (value && typeof value === "object" && !Array.isArray(value)) {
+        if (compareFields(value)) {
+          changed = true;
+        }
+      } else {
+        // Compare leaf fields
+        const savedValue = savedValues[key];
+        const currentValue = currentValues[key];
+        const initialValue = initialValues[key];
+
+        const sanitizedSaved = sanitizeValue(savedValue);
+        const sanitizedCurrent = sanitizeValue(currentValue);
+        const sanitizedInitial = sanitizeValue(initialValue);
+
+        // Compare values and log debugging information
+        // console.log(`Field: ${key}`);
+        // console.log(`  Saved: ${sanitizedSaved}`);
+        // console.log(`  Current: ${sanitizedCurrent}`);
+        // console.log(`  Initial: ${sanitizedInitial}`);
+
+        if (
+          sanitizedSaved !== sanitizedCurrent ||
+          sanitizedSaved !== sanitizedInitial
+        ) {
+          console.log(`  Field "${key}" has changed.`);
+          changed = true;
+        }
+      }
+    });
+
+    return changed;
+  };
+
+  return compareFields(sectionData);
+};
+
+
+
+
+
+const hasChangedFields = (fields, currentValues, initialValues) => {
+  return fields.some(field => {
+    // Normalize and sanitize current and initial values
+    const currentValue = sanitizeValue(currentValues[field]);
+    const initialValue = sanitizeValue(initialValues[field]);
```

Comment:
This will ensure that the currentValue and initialValue are strings, so the special handling for arrays and booleans will never be used and can be removed.